src/main/java/edu/mit/puzzle/cube/core/model/HuntStatusStore.java

```diff
@@ -327,6 +329,44 @@ public boolean setTeamProperty(
         return updates > 0;
     }
 
+    public <P extends Team.Property> boolean mutateTeamProperty(
```

Comment:
Let's say I meant concurrency / race conditions, because that's what I meant and I like to sound like a real software engineer who using the right terms. I agree that this should be at the serializable isolation level. Mostly I was concerned because even though I could see this was wrapped in a transaction, the mutation of the property being applied outside of a SQL statement could lead to an isolation violation of the kind that we are talking about now and that I didn't have the language at hand to express properly.

I further agree that literal thread safety is not at issue here.